internal/models/block.go

```diff
@@ -85,51 +91,50 @@ func (bb *BlockBSON) FromBSON() (*Block, error) {
 		return nil, fmt.Errorf("failed to parse unicityCertificate: %w", err)
 	}
 
+	var parentMerkleTreePath *api.MerkleTreePath
+	if bb.MerkleTreePath != "" {
+		hexBytes, err := api.NewHexBytesFromString(bb.MerkleTreePath)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse parentMerkleTreePath: %w", err)
+		}
+		parentMerkleTreePath = &api.MerkleTreePath{}
+		if err := json.Unmarshal(hexBytes, parentMerkleTreePath); err != nil {
+			return nil, fmt.Errorf("failed to parse parentMerkleTreePath: %w", err)
+		}
+	}
+
 	noDeletionProofHash, err := api.NewHexBytesFromString(bb.NoDeletionProofHash)
 	if err != nil {
 		return nil, fmt.Errorf("failed to parse noDeletionProofHash: %w", err)
 	}
 
-	block := &Block{
-		Index:               index,
-		ChainID:             bb.ChainID,
-		Version:             bb.Version,
-		ForkID:              bb.ForkID,
-		RootHash:            rootHash,
-		PreviousBlockHash:   previousBlockHash,
-		CreatedAt:           createdAt,
-		UnicityCertificate:  unicityCertificate,
-		NoDeletionProofHash: noDeletionProofHash,
-	}
-
-	return block, nil
+	return &Block{
+		Index:                index,
+		ChainID:              bb.ChainID,
+		ShardID:              bb.ShardID,
+		Version:              bb.Version,
+		ForkID:               bb.ForkID,
+		RootHash:             rootHash,
+		PreviousBlockHash:    previousBlockHash,
+		NoDeletionProofHash:  noDeletionProofHash,
+		CreatedAt:            api.NewTimestamp(bb.CreatedAt),
+		UnicityCertificate:   unicityCertificate,
+		ParentMerkleTreePath: parentMerkleTreePath,
+	}, nil
 }
 
 // NewBlock creates a new block
-func NewBlock(index *api.BigInt, chainID, version, forkID string, rootHash, previousBlockHash api.HexBytes) *Block {
+func NewBlock(index *api.BigInt, chainID string, shardID int, version, forkID string, rootHash, previousBlockHash, uc api.HexBytes, parentMerkleTreePath *api.MerkleTreePath) *Block {
```

Comment:
use `shardID api.ShardID` instead of `int`

Reply:
Fine. But in general I don't understand the use of api types everywhere. What if you want to change ShardID to ShardIDV2? It would be absolute nightmare to do the migration.